src/jit/codegencommon.cpp

```diff
@@ -983,20 +983,42 @@ BasicBlock* CodeGen::genCreateTempLabel()
     return block;
 }
 
-// inline
-void CodeGen::genDefineTempLabel(BasicBlock* label)
+void CodeGen::genLogLabel(BasicBlock* bb)
 {
 #ifdef DEBUG
     if (compiler->opts.dspCode)
     {
-        printf("\n      L_M%03u_" FMT_BB ":\n", Compiler::s_compMethodsCount, label->bbNum);
+        printf("\n      L_M%03u_" FMT_BB ":\n", Compiler::s_compMethodsCount, bb->bbNum);
     }
 #endif
+}
 
+// Define a label based on the current GC info tracked by the code generator.
+//
+// Notes:
+//     The label will be defined with the current GC info tracked by the code
+//     generator. When the emitter sees this label it will thus remove any temporary
+//     GC refs it is tracking in registers. For example, a call might produce a ref
+//     in RAX which the emitter would track but which would not be tracked in
+//     codegen's GC info since codegen would immediately copy it from RAX into its
+//     home.
+void CodeGen::genDefineTempLabel(BasicBlock* label)
+{
+    genLogLabel(label);
     label->bbEmitCookie =
         getEmitter()->emitAddLabel(gcInfo.gcVarPtrSetCur, gcInfo.gcRegGCrefSetCur, gcInfo.gcRegByrefSetCur);
 }
 
+// Define an inline label that does not affect the GC info.
+//
+// Notes:
+//     The emitter will continue to track GC info as if there was no label.
```

Comment:
The comment sounds confusing for me.
`The emitter will continue to track GC info as if there was no label.` <- if there is no difference with or without the label, why should we add it?  Should it be smth like `if there was no a new block`?

Looks like the many new functions are debug only and the `label` here should be `DEBUG_ARG` as well.

Reply:
The emitter continues to track GC info as if there was no label, but the label can still be used for other things than resetting GC info, like getting the address of some code instruction. That is done in my change in codegenxarch.cpp and is needed when the call could return object refs.

> Should it be smth like `if there was no a new block`?

I am not really familiar with how all the GC info works, but to me it looks like the emitter associates GC info with the label, see `emitAddLabel` that takes the GC info.

> Looks like the many new functions are debug only and the `label` here should be `DEBUG_ARG` as well.

This is not debug only, it is used when generating the call instruction.